src/makeInfinityAdder.js

```diff
@@ -4,7 +4,19 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function add(value) {
+    if(arguments.length === 0){
+      const result = sum;
+      sum = 0;
+      return result;
+    }
+
+    sum += value;
+    return add;
```

Comment:
Consider adding a comment above the `add` function to explain its behavior, especially how it handles the case when it's called without arguments. This will improve code readability and maintainability.